### PR TITLE
Add dedicated master support for ROKS 4

### PR DESCRIPTION
**Pull request by pradeepnaramaneni5**
Added dedicated master support for ROKS 4 on following components
 

- [x] cluster-version-operator-deployment
- [x] cp-operator-deployment
- [x] kube-apiserver-deployment
- [x] kube-controller-manager-deployment
- [x] kube-scheduler-deployment
- [x] oauth-apiserver-deployment
- [x] oauth-server-deployment
- [x] openshift-apiserver-deployment
- [x] cluster-policy-controller-deployment
- [x] openshift-controller-manager-deployment
- [x] manifest bootstrapper pod

### CODE REVIEW

File: assets/cluster-version-operator/cluster-version-operator-deployment.yaml [hunk 1]
```diff
@@ -30,6 +30,10 @@ spec:
                       values: ["{{ .ClusterID }}"]
                 topologyKey: "kubernetes.io/hostname"
       tolerations:
+        - key: "dedicated"
+          operator: "Equal"
+          value: "master-{{ cluster_id }}"
```
Comment (rtheis) [inline assets/cluster-version-operator/cluster-version-operator-deployment.yaml:35]: Please replace `cluster_id` with `.ClusterID`.  This is a general comment that applies to this entire PR.

**Review by rtheis — CHANGES_REQUESTED** [review]
Please also update the manifest bootstrapper pod.

File: assets/cluster-version-operator/cluster-version-operator-deployment.yaml (hunk 1 above)
Comment (pradeepnaramaneni5) [reply assets/cluster-version-operator/cluster-version-operator-deployment.yaml]: Updated. Thank you

File: assets/user-manifests-bootstrapper/user-manifests-bootstrapper-pod.yaml [hunk 2]
```diff
@@ -22,6 +22,10 @@ metadata:
   name: manifests-bootstrapper
 spec:
   tolerations:
+    - key: "dedicated"
+        operator: "Equal"
```
Comment (rtheis) [inline assets/user-manifests-bootstrapper/user-manifests-bootstrapper-pod.yaml:26]: Please fix the indentation on this and subsequent changed lines.

**Review by rtheis — CHANGES_REQUESTED** [review]
One minor comment then I think this is good to merge.  Thanks Pradeep.

**Review by rtheis — APPROVED** [review]
/lgtm
/approved

**Comment by rtheis** [comment]
**New cherry-picks:**
- https://github.com/openshift/ibm-roks-toolkit/pull/255
- https://github.com/openshift/ibm-roks-toolkit/pull/254

**Pull request merged**